app/webpacker/components/ContactsPage/SubForms/Wrt/index.jsx

```diff
@@ -21,15 +21,18 @@ export default function Wrt() {
   );
 
   const QueryForm = useMemo(() => {
-    if (!selectedQueryType) return null;
-    switch (selectedQueryType) {
-      case QUERY_TYPES_MAP.editProfile:
-        return EditProfileQuery;
-      default:
-        return OtherQuery;
-    }
+    if (!selectedQueryType || selectedQueryType === QUERY_TYPES_MAP.editProfile) return null;
+    return OtherQuery;
   }, [selectedQueryType]);
 
+  useEffect(() => {
+    if (selectedQueryType === QUERY_TYPES_MAP.editProfile) {
+      dispatch(setFormRedirection(SECTION, contactEditProfileActionUrl));
+    } else {
+      dispatch(setFormRedirection(SECTION, null));
+    }
```

Comment:
If you calculate the redirect first, and then send one dispatch with the result, you can save yourself from copy-pasting the same `dispatch` structure multiple times.

Reply:
Removed useEffect